libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs

```diff
@@ -207,9 +214,20 @@ queryCommitteeMembersState coldCredsFilter hotCredsFilter statusFilter nes = do
                 Nothing -> MemberNotAuthorized
                 Just Nothing -> MemberResigned
                 Just (Just hk) -> MemberAuthorized hk
-        -- TODO: implement after pulsing and snapshots are done
-        let nextEpochChange = NoChangeExpected
-        pure $ CommitteeMemberState hkStatus status mbExpiry nextEpochChange
+        pure $ CommitteeMemberState hkStatus status mbExpiry (nextEpochChange coldCred)
+
+      nextEpochChange :: Credential 'ColdCommitteeRole (EraCrypto era) -> NextEpochChange
+      nextEpochChange ck
+        | not inCurrent && inNext = ToBeEnacted
+        | not inNext = ToBeRemoved
+        | expiring = ToBeExpired
```

Comment:
You gonna hate me for this, but I think it could be good to add one more prediction `TermAdjusted EpochNo`. If you are not up this 
```suggestion
        | expiring = ToBeExpired
        | Just curTerm <- lookupCurrent
        , Just newTerm <- lookupFuture
        , curTerm /= newTerm = TermAdjusted newTerm
```

Reply:
That's useful information to provide in the query result! 
Are you thinking:   they're in both current and next committee, but with a different expiry epoch?

Reply:
> they're in both current and next committee, but with a different expiry epoch?

Yes. :+1: